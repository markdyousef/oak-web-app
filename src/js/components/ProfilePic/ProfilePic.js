import React, { PropTypes, Component } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';
import { SquareButton } from '../../components/shared/Button';
import { uploadImg } from '../../utils';

const Container = styled.div`
    display: flex;
    padding: 20px;
    border: 1px solid ${colors.lightGrey};
    background: ${colors.white};
    border-radius: 3px;
    flex-wrap: wrap;
`;

const Avatar = styled.div`
    height: 100px;
    width: 100px;
    margin-right: 20px;
    & img {
        width: 100%;
        height: 100%;
        border-radius: 999em;
    }
`;

const Upload = styled.div`
    width: 60%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    & p {
        font-size: 14px;
        line-height: 17px;
    }
`;

export default class ProfilePic extends Component {
    static propTypes = {
        picture: PropTypes.string,
        onChange: PropTypes.func.isRequired
    }
    static defaultProps = {
        picture: null
    }
    constructor() {
        super();
        this.state = {
            isLoading: false
        };
    }
    onClick = () => {
        this.input.value = null;
        this.input.click();
    }
    onChange = (e) => {
        e.preventDefault();
        const { onChange } = this.props;

        const file = e.target.files[0];
        const url = window.URL.createObjectURL(file);
        // // check file type
        uploadImg(file, 'avatar')
            .then(id => onChange('avatar', { id, url }))
            .catch(err => console.log(err));
    }
    render() {
        const { picture } = this.props;
        return (
            <Container>
                <Avatar>
                    <img src={picture} alt="avatar" />
                </Avatar>
                <Upload>
                    <p>
                        Upload a clear frontal face picture of yourself
                        to give your teammates a more clear identification of you.
                    </p>
                    <SquareButton
                        text="Upload a file"
                        onClick={this.onClick}
                    />
                    <input
                        type="file"
                        accept="image/*"
                        ref={(c) => { this.input = c; }}
                        onChange={this.onChange}
                        style={{ display: 'none' }}
                    />
                </Upload>
            </Container>
        );
    }
}
