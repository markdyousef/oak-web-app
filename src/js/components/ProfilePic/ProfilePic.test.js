import React from 'react';
import { shallow } from 'enzyme';
import ProfilePic from './ProfilePic';

it('renders correctly', () => {
    const props = {
        onChange: () => {}
    }
    const shallowComponent = shallow(<ProfilePic {...props} />);
    expect(shallowComponent).toMatchSnapshot();
});
