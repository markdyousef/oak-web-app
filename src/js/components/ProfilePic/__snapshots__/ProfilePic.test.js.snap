exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <img
            alt="avatar"
            src={null} />
    </styled.div>
    <styled.div>
        <label>
            Clear frontal face pictures is an important way for teammates to identify each other.
        </label>
        <Unknown
            onClick={[Function]}
            text="Upload a file"
            type="white" />
        <input
            accept="image/*"
            onChange={[Function]}
            style={
                Object {
                    "display": "none",
                  }
            }
            type="file" />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <img
                  alt="avatar"
                  src={null} />
      </styled.div>
      <styled.div>
            <label>
                  Clear frontal face pictures is an important way for teammates to identify each other.
            </label>
            <Unknown
                  onClick={[Function]}
                  text="Upload a file"
                  type="white" />
            <input
                  accept="image/*"
                  onChange={[Function]}
                  style={
                        Object {
                              "display": "none",
                            }
                  }
                  type="file" />
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ProfilePic
        onChange={[Function]}
        picture={null} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ProfilePic {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onChange": [Function],
        "onClick": [Function],
        "props": Object {
          "onChange": [Function],
          "picture": null,
        },
        "refs": Object {},
        "state": Object {
          "isLoading": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    <img
                              alt="avatar"
                              src={null} />
          </styled.div>
          <styled.div>
                    <label>
                              Clear frontal face pictures is an important way for teammates to identify each other.
                    </label>
                    <Unknown
                              onClick={[Function]}
                              text="Upload a file"
                              type="white" />
                    <input
                              accept="image/*"
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "display": "none",
                                                }
                              }
                              type="file" />
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    <img
                              alt="avatar"
                              src={null} />
          </styled.div>
          <styled.div>
                    <label>
                              Clear frontal face pictures is an important way for teammates to identify each other.
                    </label>
                    <Unknown
                              onClick={[Function]}
                              text="Upload a file"
                              type="white" />
                    <input
                              accept="image/*"
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "display": "none",
                                                }
                              }
                              type="file" />
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ProfilePic
    onChange={[Function]}
    picture={null} />,
}
`;
